src/NServiceBus.Core/Diagnostics/ActivityDecorator.cs

```diff
@@ -15,40 +14,64 @@ public static void PromoteHeadersToTags(Activity activity, Dictionary<string, st
 
             foreach (var header in headers)
             {
-                if (header.Key.StartsWith("NServiceBus.") && !IgnoreHeaders.Contains(header.Key))
+                if (HeaderMapping.TryGetValue(header.Key, out var tagName))
```

Comment:
Super-micro optimization:

It may be better to iterate over the HeaderMapping collection and do a lookup on headers. The mapping is a fixed fize and contains headers that are likely present on the message headers.

Reply:
I think in general it's safer to assume that the headers collection will be smaller than the mappings collection (at least the way it's currently defined). The mapping contains a lot of headers that I'm not sure we're using anymore at this point or might only be used in special cases. at this point it looked like the headers collection is about half the size as the mapping (based on acceptance test values).